### PR TITLE
NETOBSERV-397 Kafka TLS

### PR DESCRIPTION
**Pull request by jotak**
- Add API / config
- Read certificates / prepare TLS config for HTTP transport
- Use in ingester + encoder

### CODE REVIEW

**Comment by jotak** [comment]
PS: I'm working on adding e2e tests, but they aren't designed to work with podman at the moment, so working on that first

File: pkg/api/tls.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	}
+	if c.CACertPath != "" {
+		caCert, err := ioutil.ReadFile(c.CACertPath)
+		if err != nil {
+			return nil, err
+		}
+		tlsConfig.RootCAs = x509.NewCertPool()
+		tlsConfig.RootCAs.AppendCertsFromPEM(caCert)
+
+		if c.UserCertPath != "" && c.UserKeyPath != "" {
```
Comment (ronensc) [inline pkg/api/tls.go:45]: Does it make sense to set only one of `UserCertPath` or `UserKeyPath`? If not, should we write an error?

**Review by ronensc — APPROVED** [review]

File: pkg/pipeline/ingest/ingest_kafka.go [hunk 2]
```diff
@@ -168,13 +169,27 @@ func NewIngestKafka(params config.StageParam) (Ingester, error) {
 		commitInterval = jsonIngestKafka.CommitInterval
 	}
 
+	dialer := &kafka.Dialer{
+		Timeout:   10 * time.Second,
```
[the remark this answers is not in the document]
Comment (ronensc) [reply pkg/pipeline/ingest/ingest_kafka.go:173]: Does it make sense to make it more explicitly that we're using the default settings?
```go
dialer := &kafka.Dialer{
	Timeout:   kafka.DefaultDialer.Timeout,
	DualStack: kafka.DefaultDialer.DualStack,
}
```

File: pkg/api/tls.go (hunk 1 above)
Comment (jotak) [reply pkg/api/tls.go:46]: done, I also added some tests

File: pkg/pipeline/ingest/ingest_kafka.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (jotak) [reply pkg/pipeline/ingest/ingest_kafka.go]: +1 , done

**Pull request merged**